### PR TITLE
Allow running long tests with testOnly

### PR DESCRIPTION
**Pull request by nicolasstucki**
Improves #3194

### CODE REVIEW

**Comment by allanrenucci** [comment]
What about a solution based on [junit categories](https://github.com/junit-team/junit4/wiki/Categories)?

File: project/Build.scala [hunk 1]
```diff
@@ -465,13 +465,13 @@ object Build {
 
       test in Test := {
         // Exclude legacy tests by default
-        (testOnly in Test).toTask(" -- --exclude-categories=java.lang.Exception").value
+        (testOnly in Test).toTask(" -- --exclude-categories=java.lang.Exception,dotty.tools.dotc.SlowTests").value
```
Comment (allanrenucci) [inline project/Build.scala]: Does this mean that Drone will not run them?

Comment (nicolasstucki) [reply project/Build.scala]: Oh, it won't. I will have to find a workaround

Comment (nicolasstucki) [reply project/Build.scala]: Done

File: project/Build.scala [hunk 2]
```diff
@@ -465,13 +468,18 @@ object Build {
 
       test in Test := {
         // Exclude legacy tests by default
+        (testOnly in Test).toTask(" -- --exclude-categories=java.lang.Exception,dotty.tools.dotc.SlowTests").value
+      },
+
+      testAll in Test := {
+        // Exclude legacy tests by default
         (testOnly in Test).toTask(" -- --exclude-categories=java.lang.Exception").value
```
Comment (allanrenucci) [inline project/Build.scala]: While you're at it, could you create a meaningful category for legacy tests?

Comment (nicolasstucki) [reply project/Build.scala]: I will

**Comment by nicolasstucki** [comment]
@allanrenucci could you review the las commit

**Comment by allanrenucci** [comment]
Can we put all the categories in one file. Maybe under `compiler/test/dotty/TestCategories`.  Also I think you should be able to do `@Category(classOf[SlowTests])` instead of `@Category(Array(classOf[SlowTests]))` when a test belongs to only one category. Otherwise LGTM

**Comment by nicolasstucki** [comment]
It does not work with `@Category(classOf[SlowTests])`, probably implementation details of the JUnit plugin.

**Pull request merged**